woocommerce/woocommerce-gateway-stripe
# Make sure Stripe blocks are rendered correctly in Gutenberg

# Changes proposed in this Pull Request:

The Cart and Checkout blocks were not rendering correctly because of undefined variables while in Gutenberg, and on Cart and Checkout pages that were not set as the Cart or Checkout page in the WooCommerce Settings.

This PR fixes those issues by loading the necessary configuration when adding blocks support — and thus duplicating the configuration. We should address this duplication once we start working on #1516 .


**Before**

![image](https://user-images.githubusercontent.com/13835680/118743859-bc36ea80-b842-11eb-8b20-27c237369401.png)


**After**

![image](https://user-images.githubusercontent.com/13835680/118743885-cb1d9d00-b842-11eb-8d38-e0dada208856.png)

I've also made sure that Payment Request Buttons are loaded in preview:

![image](https://user-images.githubusercontent.com/13835680/118743916-db357c80-b842-11eb-811c-e6228f0ff778.png)


# Testing instructions

1. `git checkout trunk`
2. `npm i && npm run build:webpack`
3. Open **wp-admin > Pages > Add Page**
4. Add the Checkout Block to the Page
5. Observe that the block doesn't load properly (see screenshot above for example)
1. `git checkout fix/cart-and-checkout-block-rendering-in-gutenberg`
2. `npm i && npm run build:webpack`
3. Open **wp-admin > Pages > Add Page**
4. Add the Checkout Block to the Page
5. Observe that the block shows both a credit card form and a payment request button

-------------------
